### PR TITLE
add setting for default new file line endings

### LINKED ISSUE #5: Add setting to set default line-endings for new files
Hi, I work on Windows but always use just LF.  Since this landed all my new files are by default CRLF which is not what I want.  

Please add a setting to override the platform default for new files.  Actually, ideally, an override per file extension type :)

### PR DESCRIPTION
**Pull request by clintwood**
This PR resolves #5 by enabling setting the default line endings for **new files** as shown below.

| Setting | Effect |
| --- | --- |
| **OS Default** | Use the OS to determine which line ending to use for new files |
| **LF** | Use `'\n'` for new files |
| **CRLF** | Use `'\r\n'` for new files |

This is purely for **new files** and does not affect changes made after the new file has been created and contains text.

### CODE REVIEW

**Comment by oh-ren** [comment]
So.. Is this going in? (People want/need this.. ;))

File: lib/main.js [hunk 1]
```diff
@@ -10,6 +10,16 @@ const LineEndingRegExp = /\r\n|\n/g
 
 let disposables = null
 
+export let config = {
+  defaultNewFileLineEnding: {
+    title: 'Default line ending for new files',
+    type: 'string',
+    'default': 'OS Default',
+    enum: ['OS Default', 'LF', 'CRLF'],
+    descrition: 'Default line ending for new documents'
```
Comment (maxbrunsfeld) [inline lib/main.js]: Typo: `descrition` -> `description`.

**Comment by maxbrunsfeld** [comment]
@clintwood Thanks for adding this! Could you (or anybody who is interested) add a test for this functionality? You could expand [this test](https://github.com/atom/line-ending-selector/blob/ed18d96857565016baa35adb2615e653d6f53617/spec/line-ending-selector-spec.js#L60) to cover the cases where `line-ending-selector.defaultLineEnding` is set to `LF` and `CRLF`.

File: lib/main.js [hunk 2]
```diff
@@ -10,6 +10,16 @@ const LineEndingRegExp = /\r\n|\n/g
 
 let disposables = null
 
+export let config = {
+  defaultNewFileLineEnding: {
```
Comment (maxbrunsfeld) [inline lib/main.js]: I think `defaultNewFileLineEnding` is a bit verbose as a name. How do you feel about changing it to `defaultLineEnding`?

Comment (clintwood) [inline lib/main.js]: Sure, was to be explicit about the fact that it was just applied for new files.

Will update PR.

**Comment by clintwood** [comment]
Regarding the tests, I'm up against hard deadline ATM, and can only update tests realistically early January.  Happy to accept patches which I can check and push!

**Comment by clintwood** [comment]
Ok, I had a quick hack on the tests, please let me know if that covers the changes for this PR.

**Comment by maxbrunsfeld** [comment]
@clintwood These tests look very nice. Great work!

**Comment by clintwood** [comment]
Cool!

**Comment by maxbrunsfeld** [comment]
This will be in the next Atom release.

**Comment by PatSmuk** [comment]
Did this ever make into Atom? Would be quite useful.

**Comment by clintwood** [comment]
@Patman64, I can confirm it is in the version I'm running 1.5.3, but I would assume it landed quite a while before this version!

**Comment by PatSmuk** [comment]
OK, just found the option. For anyone who finds this in the future, you have to go to File -> Settings -> Packages -> line-ending-selector -> Settings -> Default line ending for new files. Thanks @clintwood.

**Pull request merged**